hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java

```diff
@@ -54,29 +57,34 @@
 import org.apache.ratis.retry.RetryPolicy;
 import org.apache.ratis.rpc.RpcType;
 import org.apache.ratis.rpc.SupportedRpcType;
+import org.apache.ratis.server.RaftServerConfigKeys;
 import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
 import org.apache.ratis.util.TimeDuration;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import static org.apache.hadoop.ozone.conf.DatanodeRatisServerConfig.DATANODE_RATIS_SERVER_CONFIG_PREFIX;
-
 /**
  * Ratis helper methods.
  */
 public interface RatisHelper {
   Logger LOG = LoggerFactory.getLogger(RatisHelper.class);
 
-  // Ratis Client and Grpc header regex filters.
-  String RATIS_CLIENT_HEADER_REGEX = "raft\\.client\\.([a-z\\.]+)";
-  String RATIS_GRPC_CLIENT_HEADER_REGEX = "raft\\.grpc\\.(?!server|tls)" +
-      "([a-z\\.]+)";
+  // Prefix for Ratis Server GRPC and Ratis client conf.
+  String HDDS_DATANODE_RATIS_PREFIX_KEY = "datanode.ratis.";
+  String HDDS_DATANODE_RATIS_SERVER_PREFIX_KEY = "datanode.ratis.server";
+  String HDDS_DATANODE_RATIS_CLIENT_PREFIX_KEY = "datanode.ratis.client";
+  String HDDS_DATANODE_RATIS_GRPC_PREFIX_KEY = "datanode.ratis.grpc";
+
 
-  // Ratis Server header regex filter.
-  String RATIS_SERVER_HEADER_REGEX = "datanode\\.ratis\\.raft\\.server\\" +
-      ".([a-z\\.]+)";
-  String RATIS_SERVER_GRPC_HEADER_REGEX = "datanode\\.ratis\\.raft\\.grpc\\" +
-      ".([a-z\\.]+)";
+  String RATIS_SERVER_PREFIX_KEY =
```

Comment:
Do we really need the `server` and `client` part here? Why don't we use `datanode.ratis.raft.client.*` and `datanode.ratis.raft.server.*` instead of `datanode.ratis.server.raft.server`? Is there any use case where we need to configure the client configs on the server side? (`datanode.ratis.server.raft.client.*`)

BTW, this is just a matter of taste but I would prefer to avoid substring. This seems to be more readable for me:

```
  String HDDS_DATANODE_RATIS_SERVER_TAG = "server";
  
String HDDS_DATANODE_RATIS_PREFIX_KEY = "datanode.ratis.";

  String HDDS_DATANODE_RATIS_SERVER_PREFIX_KEY =
      HDDS_DATANODE_RATIS_PREFIX_KEY + "." HDDS_DATANODE_RATIS_SERVER_TAG;


String RATIS_SERVER_PREFIX_KEY =  RATIS_SERVER_PREFIX_KEY +"." + RaftServerConfigKeys.PREFIX;
```